File: algorithms/009.palindrome-number.js
/**
 * @param {number} x
 * @return {boolean}
 */
var isPalindrome = function(x) {
	if (x < 0){
		return false;
	}
	let reverseString = x.toString().split('').reverse().join('');
	if (reverseString != x){
		return false;
	}
	return true;
};

var isPalindromeTerse = function(x) {
	return x.toString().split('').reverse().join('') === x+'';
};

File: algorithms/009.palindrome-number.go
func isPalindrome(x int) bool {
    if x < 0 {
        return false
    }

    original := x

    reversed := 0
    for x != 0 {
        reversed = reversed*10 + x % 10
        x = x / 10
    }
    
    return original == reversed
}
